dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -229,21 +228,11 @@ protected void run(@Nonnull Connector connector, @Nonnull ConnectorArguments arg
         // We had a customer request to base it on the database, but that isn't well-defined,
         // as there may be 0 or N databases in a single file.
         File outputFile = arguments.getOutputFile();
-        String outputDirectory = arguments.getOutputDirectory();
-
-        boolean isDefaultPath = "".equals(outputDirectory);
-        if (!isDefaultPath && outputFile != null) {
-            System.out.println(
-                "**********************************************************\n"
-                + "* ERROR: Using both --output and --output-dir flags is not allowed.\n"
-                + "* Please use --help for more information.\n"
-                + "**********************************************************"
-            );
-            return;
-        }
-
-        if (outputFile == null) {
-            outputFile = new File(outputDirectory, connector.getDefaultFileName(arguments.isAssessment()));
+        String defaultFileName = connector.getDefaultFileName(arguments.isAssessment());
+        if (outputFile == null || outputFile.getName().isEmpty()) {
+            outputFile = new File(defaultFileName);
+        } else if (!StringUtils.endsWithIgnoreCase(outputFile.getName(), ".zip")) {
+            outputFile = new File(outputFile.getName(), defaultFileName);
```

Comment:
One more check is needed before this line. If the path passed in the `outputFile` exists and it is a regular file (not a directory nor a directory that does not exist yet), then the dumper should fail.